woocommerce/woocommerce-android
# Check for orders locally first

⚠️ This PR depends on this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2786)

Closes: #9462

### Description
This small PR will favor the usage of the new `hasOrders` function for checking whether the store contains at least one order. This new function will rely more on the local database to check whether the store contains any order, reducing this way the number of requests we make to the API.

### Testing instructions
TC1
1. Open the MyStore screen on a store with orders
2. Change between the different granularities tabs (Today, This Week, This Month...etc.)
3. Check that no request is made to the API to check whether the store has orders 

TC1
1. Open the MyStore screen on a store without orders
2. Change between the different granularities tabs (Today, This Week, This Month...etc.)
3. Check that a request is made to the API to check whether the store has orders (previous behavior)

### Images/gif

The request to the API in the case the store has no orders should look like this:

<img width="426" alt="Screenshot 2023-07-24 at 11 58 25" src="https://github.com/woocommerce/woocommerce-android/assets/18119390/14f80409-4f23-4b3f-b474-3f775841d17b">
